gordo_components/model/multi_regressor.py

```diff
@@ -0,0 +1,142 @@
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.utils import Parallel, delayed
+
+
+class MultiTargetRegressor(BaseEstimator, RegressorMixin):
+    """
+    Support training on multi-dimensional data and predict
+    multi-dimensional output.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import RandomForestRegressor
+    >>> from gordo_components.model.models import MultiTargetRegressor
+
+    >>> # Note that y will have 4 'columns'/'targets'
+    >>> X, y = np.random.random((10, 10)), np.random.random((10, 4))
+    >>> model = MultiTargetRegressor()
+    >>> model.fit(X, y)
+    MultiTargetRegressor(base_estimator=RandomForestRegressor(bootstrap=True,
+                                                              criterion='mse',
+                                                              max_depth=None,
+                                                              max_features='auto',
+                                                              max_leaf_nodes=None,
+                                                              min_impurity_decrease=0.0,
+                                                              min_impurity_split=None,
+                                                              min_samples_leaf=1,
+                                                              min_samples_split=2,
+                                                              min_weight_fraction_leaf=0.0,
+                                                              n_estimators=10,
+                                                              n_jobs=None,
+                                                              oob_score=False,
+                                                              random_state=None,
+                                                              verbose=0,
+                                                              warm_start=False),
+                         n_jobs=-1, pre_dispatch='2*n_jobs', verbose=False)
+    >>> out = model.predict(X)
+    >>> assert out.shape == y.shape
+    """
+
+    models: List[BaseEstimator]
+
+    def __init__(
+        self,
+        base_estimator: Optional[BaseEstimator] = None,
+        n_jobs: int = -1,
+        verbose: bool = False,
+        pre_dispatch: Union[int, str] = "2*n_jobs",
+    ):
+        """
+
+        Parameters
+        ----------
+        base_estimator: BaseEstimator
+            Model which is used to predict individual targets of y.
+            Internally, this model is cloned and used for each axis=1 of y.
+        n_jobs : int or None, optional (default=None)
+            The number of CPUs to use to do the computation.
+            ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+            ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
+            for more details.
+        verbose : integer, optional
```

Comment:
Should the type not be `Optional[int]` then?

Reply:
(and similar for others)

Reply:
Is this karma? :sweat_smile: